File: ckgame-java/src/main/java/com/ckgame/politics/SchemeKind.java
package com.ckgame.politics;

/**
 * 阴谋类型。
 */
public enum SchemeKind {
    MURDER("谋杀"),
    ABDUCT("绑架"),
    FABRICATE_HOOK("伪造把柄"),
    SWAY("拉拢"),
    SEDUCE("引诱"),
    CLAIM_FABRICATION("伪造宣称");

    private final String nameZh;

    SchemeKind(String nameZh) {
        this.nameZh = nameZh;
    }

    public String nameZh() {
        return nameZh;
    }

    /** 基础每月进度。 */
    public double baseMonthlyProgress() {
        return switch (this) {
            case MURDER -> 5.0;
            case ABDUCT -> 4.0;
            case FABRICATE_HOOK -> 6.0;
            case SWAY -> 8.0;
            case SEDUCE -> 7.0;
            case CLAIM_FABRICATION -> 3.0;
        };
    }

    /** 基础隐秘值。 */
    public double secrecyBase() {
        return switch (this) {
            case MURDER -> 40.0;
            case ABDUCT -> 35.0;
            case FABRICATE_HOOK -> 50.0;
            case SWAY -> 80.0;
            case SEDUCE -> 60.0;
            case CLAIM_FABRICATION -> 70.0;
        };
    }
}

File: ckgame-ts/src/politics/SchemeKind.ts
/**
 * Scheme (conspiracy) types.
 */
export enum SchemeKind {
  MURDER = 'MURDER',
  ABDUCT = 'ABDUCT',
  FABRICATE_HOOK = 'FABRICATE_HOOK',
  SWAY = 'SWAY',
  SEDUCE = 'SEDUCE',
  CLAIM_FABRICATION = 'CLAIM_FABRICATION',
}

const _nameZh: Record<SchemeKind, string> = {
  [SchemeKind.MURDER]: '谋杀',
  [SchemeKind.ABDUCT]: '绑架',
  [SchemeKind.FABRICATE_HOOK]: '伪造把柄',
  [SchemeKind.SWAY]: '拉拢',
  [SchemeKind.SEDUCE]: '引诱',
  [SchemeKind.CLAIM_FABRICATION]: '伪造宣称',
};

const _baseMonthlyProgress: Record<SchemeKind, number> = {
  [SchemeKind.MURDER]: 5.0,
  [SchemeKind.ABDUCT]: 4.0,
  [SchemeKind.FABRICATE_HOOK]: 6.0,
  [SchemeKind.SWAY]: 8.0,
  [SchemeKind.SEDUCE]: 7.0,
  [SchemeKind.CLAIM_FABRICATION]: 3.0,
};

const _secrecyBase: Record<SchemeKind, number> = {
  [SchemeKind.MURDER]: 40.0,
  [SchemeKind.ABDUCT]: 35.0,
  [SchemeKind.FABRICATE_HOOK]: 50.0,
  [SchemeKind.SWAY]: 80.0,
  [SchemeKind.SEDUCE]: 60.0,
  [SchemeKind.CLAIM_FABRICATION]: 70.0,
};

export function schemeKindNameZh(kind: SchemeKind): string {
  return _nameZh[kind] ?? kind;
}

export function schemeKindBaseMonthlyProgress(kind: SchemeKind): number {
  return _baseMonthlyProgress[kind] ?? 5.0;
}

export function schemeKindSecrecyBase(kind: SchemeKind): number {
  return _secrecyBase[kind] ?? 50.0;
}
